dialogue-clients/src/main/java/com/palantir/dialogue/clients/DnsSupport.java

```diff
@@ -179,10 +196,10 @@ static ImmutableList<TargetUri> getTargetUris(
                             "Resolved no addresses for host '{}' of service '{}'",
                             UnsafeArg.of("host", host),
                             SafeArg.of("service", serviceNameForLogging));
-                }
-                for (InetAddress addr : resolvedAddresses) {
-                    targetUris.add(
-                            TargetUri.builder().uri(uri).resolvedAddress(addr).build());
+                } else {
+                    TargetUri.Builder builder = TargetUri.builder().uri(uri);
+                    targetUris.addAll(Collections2.transform(resolvedAddresses, addr -> builder.resolvedAddress(addr)
+                            .build()));
```

Comment:
I'm not a big fan of the partial-builder-as-factory pattern, since sometimes folks add protections later on to prevent interactions after `build()` is called. It shouldn't be an issue here since. the TargetUri type is defined within this repo, so I don't have a terribly strong preference

Reply:
Perhaps we add an efficient `TargetUri.of(uri, address)` factory method to use instead? That way we bypass the builder entirely.

Reply:
good call, updated